src/components/Items/Scorecard/Scorecard.vue

```diff
@@ -0,0 +1,252 @@
+    window.addEventListener("resize", this.handleScreenSizeChange);
+  },
+  mounted() {},
+  unmounted() {
+    window.removeEventListener("resize", this.handleScreenSizeChange);
+  },
+  computed: {
+    circularProgressRadius() {
+      // reactively control the radius of the circular progress bar
+      // according to the screen width
+      if (!this.isPortrait) {
+        if (this.innerWidth >= 1200) return 150;
+        else if (this.innerWidth < 1200 && this.innerWidth >= 1024) return 120;
+        else if (this.innerWidth < 1024 && this.innerWidth >= 768) return 100;
+        else if (this.innerWidth < 768 && this.innerWidth >= 640) return 80;
+        return 50;
+      } else return 100;
+    },
+    circularProgressStroke() {
+      // reactively control the stroke of the circular progress bar
+      // according to the screen width
+      if (!this.isPortrait) {
+        if (this.innerWidth >= 1200) return 20;
+        else if (this.innerWidth < 1200 && this.innerWidth >= 1024) return 18;
+        else if (this.innerWidth < 1024 && this.innerWidth >= 768) return 15;
+        else if (this.innerWidth < 768 && this.innerWidth >= 640) return 12;
+        return 8;
+      } else return 15;
+    },
+    watchAgainButtonTitleConfig() {
+      // config for the text of the watch again button
+      return {
+        value: this.$t("player.scorecard.buttons.watchAgain"),
+        class: "text-white text-sm sm:text-base lg:text-lg font-bold",
+      };
+    },
+    watchAgainButtonClass() {
+      // class for the watch again button
+      return "bg-primary hover:bg-primary-hover p-1 pl-4 pr-4 sm:p-2 sm:pl-10 sm:pr-10 lg:p-4 lg:pl-10 lg:pr-10 rounded-md shadow-xl disabled:opacity-50 disabled:pointer-events-none";
+    },
+  },
+  methods: {
+    ...Utilties,
+    handleScreenSizeChange() {
+      // invoked when the screen size is changing
+      this.innerWidth = window.innerWidth;
+
+      // re-render all components that are using the reRenderKey
+      // here - Scorecard gets rerender -- to properly place the progress bar
+      this.reRenderKey = !this.reRenderKey;
+    },
+    throwConfetti() {
+      // keep throwing some confetti until our specified time runs out
+      var end = Date.now() + 10 * 1000;
+      var colors = ["#ff718d", "#fdff6a"];
+
+      const frame = () => {
+        confettiHandler({
+          particleCount: 2,
+          angle: 60,
+          spread: 55,
+          origin: { x: 0 },
+          colors: colors,
+        });
+        confettiHandler({
+          particleCount: 2,
+          angle: 120,
+          spread: 55,
+          origin: { x: 1 },
+          colors: colors,
+        });
+
+        if (Date.now() < end) {
+          // store the animation frame request in a variable
+          // so we can cancel it later on
+          this.req = requestAnimationFrame(frame);
+        }
+      };
+      frame();
+    },
+    watchAgain() {
+      // when watch again is clicked
```

Comment:
rather than writing when the method is triggered, let's write what the method does.